flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -910,40 +930,25 @@ public ExecutionConfig getExecutionConfig() {
 		final int inputParallelism = getTransformation().getParallelism();
 		final AssignerWithPeriodicWatermarks<T> cleanedAssigner = clean(timestampAndWatermarkAssigner);
 
-		TimestampsAndPeriodicWatermarksOperator<T> operator =
-				new TimestampsAndPeriodicWatermarksOperator<>(cleanedAssigner);
+		final WatermarkStrategy<T> wms = new AssignerWithPeriodicWatermarksAdapter.Strategy<>(cleanedAssigner);
+		final TimestampsAndWatermarksOperator<T> operator = new TimestampsAndWatermarksOperator<>(wms);
```

Comment:
how about we redirect to the non deprecated method? The only downside is that the closure cleaner will be applied on the AssignerWithPeriodicWatermarksAdapter.Strateg but I think its fine.

Reply:
👌